ServerCore/Pages/Events/CreateDemo.cshtml.cs

```diff
@@ -57,194 +61,223 @@ public async Task<IActionResult> OnPostAsync()
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = intermediate, Prerequisite = start });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = hard, Prerequisite = easy });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = hard, Prerequisite = intermediate });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = meta, Prerequisite = easy });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = meta, Prerequisite = intermediate });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = meta, Prerequisite = hard });
+                _context.Prerequisites.Add(new Prerequisites() { Puzzle = other, Prerequisite = start });
+
+                await _context.SaveChangesAsync();
+
+                //
+                // Create teams. Can we add players to these?
+                //
+                Team team1 = new Team { Name = "Team Bugs", Event = Event };
+                _context.Teams.Add(team1);
+
+                Team team2 = new Team { Name = "Team Babs", Event = Event };
+                _context.Teams.Add(team2);
+
+                Team team3 = new Team { Name = "Team Buster", Event = Event };
+                _context.Teams.Add(team3);
+
+                Team teamLoneWolf = null;
+                if (AddCreatorToLoneWolfTeam)
+                {
+                    teamLoneWolf = new Team { Name = "Lone Wolf", Event = Event };
+                    _context.Teams.Add(teamLoneWolf);
+                }
+
+                var demoCreatorUser = await PuzzleUser.GetPuzzleUserForCurrentUser(_context, User, _userManager);
+                if (demoCreatorUser != null)
+                {
+                    //
+                    // Event admin/author
+                    //
+                    _context.EventAdmins.Add(new EventAdmins() { Event = Event, Admin = demoCreatorUser });
+                    _context.EventAuthors.Add(new EventAuthors() { Event = Event, Author = demoCreatorUser });
+
+                    //
+                    // Puzzle author (for Thumper module only)
+                    //
+                    _context.PuzzleAuthors.Add(new PuzzleAuthors() { Puzzle = easy, Author = demoCreatorUser });
+                    _context.PuzzleAuthors.Add(new PuzzleAuthors() { Puzzle = intermediate, Author = demoCreatorUser });
+                    _context.PuzzleAuthors.Add(new PuzzleAuthors() { Puzzle = hard, Author = demoCreatorUser });
+                    _context.PuzzleAuthors.Add(new PuzzleAuthors() { Puzzle = meta, Author = demoCreatorUser });
+                }
+
+                // TODO: Files (need to know how to detect whether local blob storage is configured)
+                // Is there a point to adding Feedback or is that quick/easy enough to demo by hand?
+
+                await _context.SaveChangesAsync();
+
+                if (teamLoneWolf != null)
```

Comment:
Is there a reason this isn't just part of the if statement on line 225? Seems like it could be done there - not sure what I'm missing

Reply:
It needs to be somewhere after a SaveChangesAsync() in order to be valid - that's what gets the team ID set up.